workers/unity/Assets/Playground/Editor/Prefabs/PrefabPreprocessor.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Improbable.Gdk.Core.GameObjectRepresentation;
+using UnityEditor;
+using UnityEditor.Build;
+using UnityEditor.Build.Reporting;
+using UnityEngine;
+using Object = UnityEngine.Object;
+
+namespace Playground.Editor
+{
+    [InitializeOnLoad]
+    public class PrefabPreprocessor : IPreprocessBuildWithReport
+    {
+        // Needed for IPreprocessBuildWithReport
+        public int callbackOrder => 0;
+
+        static PrefabPreprocessor()
+        {
+            EditorApplication.playModeStateChanged += OnPlaymodeStateChanged;
+        }
+
+        void IPreprocessBuildWithReport.OnPreprocessBuild(BuildReport report)
+        {
+            PreprocessPrefabs();
+        }
+
+        private static void OnPlaymodeStateChanged(PlayModeStateChange playModeStateChange)
+        {
+            if (playModeStateChange == PlayModeStateChange.ExitingEditMode)
+            {
+                PreprocessPrefabs();
+            }
+        }
+
+        private static void PreprocessPrefabs()
+        {
+            var assetsUpdated = false;
+
+            var allPrefabObjectsInResources = AssetDatabase.FindAssets("t:Prefab")
+                .Select(AssetDatabase.GUIDToAssetPath)
+                .Where(assetPath => assetPath.Contains("Resources"))
+                .Select(AssetDatabase.LoadAssetAtPath<GameObject>)
+                .Where(prefabObject => prefabObject != null);
```

Comment:
Minor point - might be worth adding ToArray at the end of this to prevent future work doing unexpected things. If left as an IEnumerable it will actually evaluate the entire LINQ expression every time the variable is accessed. Has caught people out before when trying to augment these kinds of scripts.

Reply:
👍